src/vellum/workflows/nodes/displayable/tool_calling_node/composio_service.py

```
@@ -13,65 +15,58 @@ class ConnectionInfo:
     integration_name: str
     created_at: str
     updated_at: str
-    status: str = "ACTIVE"  # TODO: Use enum if we end up supporting integrations that the user has not yet connected to
+    status: str = "ACTIVE"
 
 
-class ComposioAccountService:
-    """Manages user authorized connections using composio-client"""
+class ComposioService:
+    """Composio API client for managing connections and executing tools"""
 
     def __init__(self, api_key: str):
-        self.client = ComposioClient(api_key=api_key)
+        self.api_key = api_key
+        self.base_url = "https://backend.composio.dev/api/v3"
+
+    def _make_request(
+        self, endpoint: str, method: str = "GET", params: Optional[dict] = None, json_data: Optional[dict] = None
+    ) -> dict:
+        """Make a request to the Composio API"""
+        headers = {
+            "x-api-key": self.api_key,
+            "Content-Type": "application/json",
+        }
+
+        url = f"{self.base_url}{endpoint}"
+
+        try:
+            if method == "GET":
+                response = requests.get(url, headers=headers, params=params or {}, timeout=30)
+            elif method == "POST":
+                response = requests.post(url, headers=headers, json=json_data or {}, timeout=30)
+            else:
+                raise ValueError(f"Unsupported HTTP method: {method}")
+
+            response.raise_for_status()
+            return response.json()
+        except Exception as e:
+            logger.error(f"Composio API request failed: {e}")
```

[q] should we raise node exception here instead of logger error?

---

Good call - swapping this out for NodeException makes the errors within the node visible